Added a GAP notebook example.

rbehrends opened the pull request:
This is a very mechanical translation of the [Rubik's Cube Example](https://www.gap-system.org/Doc/Examples/rubik.html), meant to address issue #3. It currently still misses text annotations and has its share of ugliness.

I also noticed that the pre-image computations gave different results, but haven't yet looked into the underlying causes.

fingolfin commented:
Thanks! Some notes:
- we should add constructors to GAP.jl, so that e.g. `BigInt(some_gap_int)` works (see also https://github.com/oscar-system/GAP.jl/pull/323)
- I already told you about the `@gap` macro on Slack, you can use that here
- we should add something like `Base.in(x::GAP.GapObj, y::GAP.GapObj) = GAP.Globals.in(x, y)` to `GAP.jl` (I'll submit a PR for that shortly)
- once https://github.com/oscar-system/Oscar.jl/pull/84 is merged, we should make a variant of this notebook which uses it -- the two notebooks then could reference each other and be used to showcase the differences

ThomasBreuer commented on RubiksCubeGAP.ipynb:
```diff
@@ -0,0 +1,984 @@
+    }
+   ],
+   "source": [
+    "orbits = G.Orbits(cube, to_gap(Vector(1:48)))"
```
`Vector` can be omitted: `to_gap(1:48)` gets translated to `[ 1 .. 48 ]`, which is fine for `G.Orbits`.

rbehrends replied:
Done.

ThomasBreuer commented on RubiksCubeGAP.ipynb:
```diff
@@ -0,0 +1,984 @@
+    }
+   ],
+   "source": [
+    "f = @gap(\"FreeGroup(\\\"t\\\", \\\"l\\\", \\\"f\\\", \\\"r\\\", \\\"e\\\", \\\"b\\\")\")"
```
`G.FreeGroup( g"t", g"l", g"f", g"r", g"e", g"b" )` is more readable.

rbehrends replied:
I've actually changed that to triple quotes in the meantime, but illustrating the use of GAP strings (which I didn't know about) would be more interesting.

fingolfin replied:
Yet another variant: `G.FreeGroup(to_gap(["t", "l", "f", "r", "e", "b"]))`. All of these have pros and cons. In the end, it depends on taste.

ThomasBreuer commented on RubiksCubeGAP.ipynb:
```diff
@@ -0,0 +1,984 @@
+    }
+   ],
+   "source": [
+    "pre1 = G.PreImagesRepresentative(hom, GAP.call_gap_func(@gap(\"x -> x.1\"), z) )"
```
For the moment, `G.PreImagesRepresentative(hom, G.GeneratorsOfGroup(z)[1])` would look more natural.
The GAP syntax `z.1` belongs to a special method only for groups, but if it would be regarded as interesting, one could make it available also in Julia. (I would not be in favor of this.)

rbehrends replied:
Hmm. I've figured out in the meantime that GAP functions can actually be called directly, e.g `@gap("x -> x + 1")(z)`; this might be illustrative, but I think your approach is more idiomatic. I'll think about that.

fingolfin replied:
You can also just write `pre1 = G.PreImagesRepresentative(hom, z.:1)`

rbehrends commented:
I can't judge myself whether this is useful for a wider audience; I simply picked the first large example on the GAP website and adapted it. I figured that even if it is not of interest for the examples on the OSCAR website (which I think is more about adding examples relevant to the overall TRR project), it can still serve educational purposes.

I'll be happy to adapt the comments from the original code if people think this example is useful. I simply wanted to get something workable out first.

fingolfin commented:
I think it's useful, and adding the comments would be good.

rbehrends commented:
I've added descriptive text and added some more comments on the specifics of using this with GAP.jl. The notebook can be viewed [here](https://github.com/rbehrends/OSCARBinder/blob/gap-example/RubiksCubeGAP.ipynb) directly.

fingolfin commented:
@rbehrends Can you please update this, now that GAP.jl 0.4.0 is out (so e.g. `in` is now available)

fingolfin commented:
Also you shouldn't need `big_int` anymore, `big` should work instead
